### PR TITLE
Return all error messages instead of throwing on the first error in linker tests

### PR DESCRIPTION
**Pull request by jtschuster**
Debugging the linker tests would be easier if we could see all the missing/extra members/warnings, but the tests currently throw an exception at the first unexpected thing. Instead, we could return a sequence of error messages to make it easier to see a pattern in the missing members and make debugging easier.

### CODE REVIEW

**Review by sbomer — COMMENTED** [review]
Overall looks good, but we should try to keep the structure of the nativeaot AssemblyChecker similar. Would you mind making the same kind of change there too?

**Comment by jtschuster** [comment]
> Overall looks good, but we should try to keep the structure of the nativeaot AssemblyChecker similar. Would you mind making the same kind of change there too?

Made the same changes for NativeAOT. A few places still assert for invalid test parameters or failing to resolve a reference, but all the `Verify...` methods return a list of error messages and concat them before throwing.

**Review by sbomer — APPROVED** [review]
LGTM once it builds and tests pass. Thanks!

File: src/tools/illink/test/Mono.Linker.Tests/TestCasesRunner/AssemblyChecker.cs [hunk 1]
```diff
@@ -508,51 +574,70 @@ void VerifyMemberBackingField (IMemberDefinition src, TypeDefinition linkedType)
...
+				yield return err;
+			foreach (var err in VerifyLocals (src, linked))
+				yield return err;
 		}
 
-		protected static void VerifyInstructions (MethodDefinition src, MethodDefinition linked)
+		protected static IEnumerable<string> VerifyInstructions (MethodDefinition src, MethodDefinition linked)
 		{
-			VerifyBodyProperties (
+			let errs = VerifyBodyProperties (
```
[the remark this answers is not in the document]
Comment (jtschuster) [reply src/tools/illink/test/Mono.Linker.Tests/TestCasesRunner/AssemblyChecker.cs:631]: Oops, been using too much F# in my free time...

**Comment by jtschuster** [comment]
Failures are known.

**Pull request merged**